traitsui/testing/tester/qt4/implementation/range_editor.py

```diff
@@ -36,10 +41,32 @@ def resolve_location_slider(wrapper, location):
     if location == locator.WidgetType.textbox:
         return LocatedTextbox(textbox=wrapper.target.control.text)
     if location in [locator.WidgetType.slider]:
-        raise NotImplementedError(
-            f"Logic for interacting with the {location}"
-            " has not been implemented."
-        )
+        return LocatedSlider(slider=wrapper.target.control.slider)
+    raise ValueError(
+        f"Unable to resolve {location} on {wrapper.target}."
+        " Currently supported: {locator.WidgetType.textbox} and"
+        " {locator.WidgetType.slider}"
+    )
+
+
+def resolve_location_spin(wrapper, location):
+    """ Solver from a UIWrapper wrapped Range Editor to a LocatedTextbox
+    containing the textbox of interest.
+
+    If there are any conflicts, an error will occur.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        Wrapper containing the SimpleSpin Range Editor target.
+    location : locator.WidgetType
+        The location we are looking to resolve.
+    """
+    if location == locator.WidgetType.textbox:
+        # FIX ME: THIS CODE DOESNT BELONG HERE
+        textbox = wrapper.target.control.lineEdit()
+        textbox.end(False)
+        return LocatedTextbox(textbox=textbox)
```

Comment:
This was another insertion point issue.  It seems qts default behavior was different here for whatever reason.  This should probably be moved to inside helpers though

Reply:
This kind of problem was seen before, and addressed by adding:
```
if not control.hasFocus():
        key_click(widget=control, key="End", delay=0)
``` 
to `keey_sequence_textbox`.  However, adding the same thing to `key_click_qwidget` does not seem to solve this problem here, and we also shouldn't call the method used above: `textbox.end(False)` because this function is supposed to work with general qwidgets, and those may not have the .end() method.

Reply:
maybe this code is fine where it is...?